docs/release-procedure.md

```diff
@@ -95,7 +119,14 @@ to test that the wheels were uploaded correctly.
 
 ### Github
 
-After all is said and done and pushed to PyPI, open Github Releases page and create a new release. It should be enough to just specify "release from tag" and point it to newly created `X.Y.Z` tag (ensure the release title reads `Modin X.Y.Z`) - Github should pull everything including release text from your tag annotations.
+Once the tag has been published, we need to make the release on GitHub. Go to the
+[Release page](https://github.com/modin-project/modin/releases), and click on `Draft a new release`.
+Choose the tag you made above from the dropdown menu, and copy paste the name of the release 
+in the `Release title` box. Next, copy paste the release notes from above into the box labelled
+`Describe this release`. This will ensure that the release notes on GitHub are Markdown formatted.
+
+Double check that everything looks good by clicking `Preview`, and then hit the green `Publish release`
+button!
```

Comment:
should we add a line on publishing that release in modin-slack `#general` channel?

Reply:
Sgtm!